examples/provider/bid/audience-network/src/audience-network-bid-delegate.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @file Audience Network BidDelegate.
+ */
+import 'whatwg-fetch';
+import isArray from 'lodash/isArray';
+import isFunction from 'lodash/isFunction';
+import isObject from 'lodash/isObject';
+import isString from 'lodash/isString';
+
+const name = 'audience-network';
+
+/**
+ * Does this slot contain valid parameters?
+ * @param {Object} slot
+ * @returns {Boolean}
+ */
+const validateSlot = slot =>
+  isObject(slot) &&
+  isString(slot.name) &&
+  isString(slot.elementId) &&
+  isArray(slot.sizes) &&
+  slot.sizes.length > 0 &&
+  isFunction(slot.getParam) &&
+  isObject(slot.getParam(name)) &&
+  isString(slot.getParam(name).placementId);
+
+/**
+ * Does this slot contain valid sizes?
+ * @param {Object} slot
+ * @returns {Boolean}
+ */
+const validateSlotSizes = slot => {
+  slot.sizes = slot.sizes.map(flattenSize);
```

Comment:
The `slot` here is a reference i.e. slot sizes configuration passed to `addSlot()`, as in the README example `sizes: [[300, 250]]`, will be redefined.

At this time, Pubfood does not pass cloned Slot objects to delegates. This function, setting of `adformats` and the `pushBid` argument for sizes will need to reflect this.